platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -518,21 +521,23 @@ public void setApplicationPlugins(List<ApplicationPlugin> applicationPlugins) {
    *
    * @return the service or <code>null</code> if missing.
    */
+  @Nullable
   final MetaTypeService getMetaTypeService() {
-    if (serviceTracker == null) {
-      BundleContext context = getContext();
-      serviceTracker = new ServiceTracker<Object, Object>(context, META_TYPE_NAME, null);
+    BundleContext context = getContext();
+
+    if (serviceTracker == null && context != null) {
+      serviceTracker = new ServiceTracker<>(context, META_TYPE_NAME, null);
       serviceTracker.open();
     }
-    return (MetaTypeService) serviceTracker.getService();
+    return (serviceTracker != null) ? (MetaTypeService) serviceTracker.getService() : null;
   }
 
   protected BundleContext getContext() {
     Bundle cxfBundle = FrameworkUtil.getBundle(ApplicationServiceBean.class);
     if (cxfBundle != null) {
       return cxfBundle.getBundleContext();
     }
-    return null;
+    throw new IllegalStateException("Could not get context from cxfBundle");
```

Comment:
Could cxfBundle.getBundleContext() return null still? I think we want these methods to never return null. Could remove your context null check up above if we change this to throw the IllegalStateException if the result returned would be null.

Reply:
Good catch